src/features/script-load.js

```diff
@@ -5,18 +5,68 @@ import { hasDocument } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 import { errMsg } from '../err-msg.js';
 
+if (hasDocument) {
+  window.addEventListener('error', function (evt) {
+    lastWindowErrorUrl = evt.filename;
+    lastWindowError = evt.error;
+  });
+  var baseOrigin = location.origin
```

Comment:
It's a bit unusual to rely on variable hoisting outside of the current block, especially since it doesn't work with let/const and we've run into issues before when switching between the two.

It appears that `baseOrigin` is only used in one place - why not just use `location.origin` directly on line 23? The `createScript` function only is called in the browser, not NodeJS, so I don't think we'll run into issues where createScript is called but location is undefined?

Reply:
`location.origin` is effectively a getter as far as I'm aware, so in theory it's a perf saving.

Reply:
People do find `var` confusing yes. But personally I find eg:

```js
try {
  var x = thing();
}
catch {}
x.stuff();
```

and other blocks a useful pattern.